### PR TITLE
FluxTubes: some improvements in shapes and material

### PR DESCRIPTION
**Pull request by AHaumer**
Checked all shapes, FluxTubes and QuasiStatic.FluxTubes in parallel

### CODE REVIEW

File: Modelica/Magnetic/QuasiStatic/FluxTubes/Shapes/FixedShape/HollowCylinderCircumferentialFlux.mo [hunk 1]
```diff
@@ -15,7 +15,7 @@ model HollowCylinderCircumferentialFlux "Hollow cylinder with circumferential fl
   parameter SI.Angle alpha=pi/2 "Angle of cylinder section"
     annotation (Dialog(group="Fixed geometry"));
 equation
-  A = l*(r_o - r_i) "Area at arithmetic mean radius for calculation of average flux density";
+  A = (r_o - r_i)*l;
```
Comment (dietmarw) [inline Modelica/Magnetic/QuasiStatic/FluxTubes/Shapes/FixedShape/HollowCylinderCircumferentialFlux.mo:18]: Again, why was the description removed?

**Review by dietmarw — COMMENTED** [review]
The new images have no longer a transparent background. Was that on purpose? This will look strange when compared to the other images that are still transparent.

**Comment by dietmarw** [comment]
@AHaumer I also had to revert your "???" commit once again here. Please follow the instruction I gave you of basing your future PRs on the current master (rather than keeping to merge the current master into your "broken" local master with that "???" commit).

**Comment by AHaumer** [comment]
> The new images have no longer a transparent background. Was that on purpose? This will look strange when compared to the other images that are still transparent.

Sorry this was by mistake, I'll correct that. I suppose you mena the images for HollowCylinder RadialFlux and Axial Flux?

**Comment by AHaumer** [comment]
> @AHaumer I also had to revert your "???" commit once again here. Please follow the instruction I gave you of basing your future PRs on the current master (rather than keeping to merge the current master into your "broken" local master with that "???" commit).

Sorry I didn't get that. Where did you give these instructions (PM or in a PR)? I don't know how to fix my broken master (and I have no idea how it got broken). Thanks in advance @dietmarw !

**Comment by AHaumer** [comment]
> > The new images have no longer a transparent background. Was that on purpose? This will look strange when compared to the other images that are still transparent.
> 
> Sorry this was by mistake, I'll correct that. I suppose you mean the images for HollowCylinder RadialFlux and Axial Flux?

I'm a little bit confused ... to me all images have a white background, also the older ones?
BTW, I don't see the difference between white and transparent background (at least in Dymola, in the parameter menu).

**Comment by dietmarw** [comment]
> I'm a little bit confused ... to me all images have a white background, also the older ones? BTW, I don't see the difference between white and transparent background (at least in Dymola, in the parameter menu).
@AHaumer Have a look at the changed files here and scroll right to the bottom where GitHub shows you the changes to the images. As you can see the original had a checkered background (symbolising the transparency) the new ones have a plain white background which is not transparent. Dymola would only show you this if you started it in dark mode (`-dark` or `/dark`)
image

before it looked like this:
image

**Comment by AHaumer** [comment]
@dietmarw sorry I didn't see that. Also two other images have white instead of transparent background - corrected all four. 
BTW please point out once again for me how I can repair my "broken" master. Thanks in advance!

**Comment by dietmarw** [comment]
@AHaumer Simply checkout your local master and reset it to upstream master and then you can create your topic branches for future PRs based on that again.
```
git checkout master
git reset --hard upstream/master  # if your remote for MSL is called upstream
```

**Comment by HansOlsson** [comment]
> @dietmarw sorry I didn't see that. Also two other images have white instead of transparent background - corrected all four. BTW please point out once again for me how I can repair my "broken" master. Thanks in advance!

When my master branch had been messed up I followed https://stackoverflow.com/questions/42332769/how-do-i-reset-the-git-master-branch-to-the-upstream-branch-in-a-forked-reposito and did:
```
git checkout master
git reset upstream/master
git pull --rebase upstream master
git push origin master --force
```
I don't know the difference wrt dietmar's variant.

**Comment by dietmarw** [comment]
Sorry I had to remove @HansOlsson 's comment since that is plain wrong and would damage the master.

**Comment by HansOlsson** [comment]
> Sorry I had to remove @HansOlsson 's comment since that is plain wrong and would damage the master. You _never_ force push to master on this project.

I obviously force push to my local master (origin - as shown in the command); not the project master (upstream).
Without that force push some of the issues will not disappear.

**Comment by dietmarw** [comment]
@HansOlsson The problem is that depending on the setup some people have origin pointing to the upstream (i.e., Modelica/ModelicaStandardLibrary) and not their fork. That's why it is problematic. To be honest the master branch on ones personal fork has no practical use since you never going (well should not) create any PR based on that.

**Comment by AHaumer** [comment]
> @AHaumer Simply checkout your local master and reset it to upstream master and then you can create your topic branches for future PRs based on that again.
> 
> ```
> git fetch --all
> git checkout master
> git reset --hard upstream/master  # if your remote for MSL is called upstream
> ```
> 
> (edited to also fetch the latest states of remote first)

Well I did that:
Your branch and 'origin/master' have diverged,
and have 5 and 4 different commits each, respectively.
(use "git pull" to merge the remote branch into yours)
I'm afraid that might have fixed the local master but not the origin.

**Comment by dietmarw** [comment]
Well good you solved it. As I said, your origin/master has no practical use what so ever. So having that out of sync would not cause any problems. But you can of course from time to time do a 
```
git checkout master
git reset --hard upstream/master 
git push --force origin/master
```
if you want (just to keep your forked master on GitHub in sync). Up to you. PR will be generated from topic branches anyway and those should always start from your local master that has been reset to upstream/master as a first step. You then push that topic branch and create the PR.

**Comment by AHaumer** [comment]
> Well good you solved it. As I said, your origin/master has no practical use what so ever. So having that out of sync would not cause any problems. But you can of course from time to time do a
> 
> ```
> git checkout master
> git reset --hard upstream/master 
> git push --force origin/master
> ```
> 
> if you want (just to keep your forked master on GitHub in sync). Up to you. PR will be generated from topic branches anyway and those should always start from your local master that has been reset to upstream/master as a first step. You then push that topic branch and create the PR.

Thanks for your help and explanations @dietmarw 
Sometimes its not really clear for me how git behaves.

**Review by dietmarw — APPROVED** [review]
Looks good to me.
@AHaumer maybe you could also add the source files of the images next to the pngs?

**Comment by beutlich** [comment]
> Looks good to me. @AHaumer maybe you could also add the source files of the images next to the pngs?

You still can add them to this repo and exclude them from packaging in the attributes files like done here:

https://github.com/modelica/ModelicaStandardLibrary/blob/569c5253c46408343a01166715387d417821c04a/.gitattributes#L40-L42

See also #2603.

File: Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/NickelIron/Permenorm3601K3.mo [hunk 2]
```diff
@@ -1,12 +1,14 @@
 within Modelica.Magnetic.FluxTubes.Material.SoftMagnetic.NickelIron;
 record Permenorm3601K3 "PERMENORM 3601 K3 (36% NiFe)"
   extends FluxTubes.Material.SoftMagnetic.BaseData(
+    label="Permenorm 3601 K3",
```
Comment (christiankral) [inline Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/NickelIron/Permenorm3601K3.mo:4]:
```suggestion
    label="PERMENORM 3601 K3",
```
See https://vacuumschmelze.de/Unternehmen/Qualitat-und-Sicherheit/ALT%20Informationsblaetter%20INKL%20DAUERMAGNETE/PERMENORM

File: Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/PureIron/VacoferS2.mo [hunk 3]
```diff
@@ -1,12 +1,14 @@
 within Modelica.Magnetic.FluxTubes.Material.SoftMagnetic.PureIron;
 record VacoferS2 "VACOFER S2 (99.95% Fe)"
   extends FluxTubes.Material.SoftMagnetic.BaseData(
+    label="Vacofer S2",
```
Comment (christiankral) [inline Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/PureIron/VacoferS2.mo:4]:
```suggestion
    label="VACOFER S2",
```
See https://vacuumschmelze.com/Company/Quality-and-Safety/Information-for-Manufactured-Articles/VACOFER

File: Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/Steel/Steel_9SMn28K.mo [hunk 4]
```diff
@@ -1,12 +1,14 @@
 within Modelica.Magnetic.FluxTubes.Material.SoftMagnetic.Steel;
 record Steel_9SMn28K "9SMn28k (1.0715)"
   extends FluxTubes.Material.SoftMagnetic.BaseData(
+    label="9SMn28k",
```
Comment (christiankral) [inline Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/Steel/Steel_9SMn28K.mo:4]:
```suggestion
record Steel_9SMn28K "9SMn28K (1.0715)"
  extends FluxTubes.Material.SoftMagnetic.BaseData(
    label="9SMn28K",
```

The are both spellings around, with `k` and `K`. I am following the record name here...

File: Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/CobaltIron/Vacoflux50.mo [hunk 5]
```diff
@@ -1,12 +1,14 @@
 within Modelica.Magnetic.FluxTubes.Material.SoftMagnetic.CobaltIron;
 record Vacoflux50 "Vacoflux 50 (50% CoFe)"
   extends FluxTubes.Material.SoftMagnetic.BaseData(
+    label="Vacoflux 50",
```
Comment (christiankral) [inline Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/CobaltIron/Vacoflux50.mo:4]:
```suggestion
record Vacoflux50 "VACOFLUX 50 (50% CoFe)"
  extends FluxTubes.Material.SoftMagnetic.BaseData(
    label="VACOFLUX 50",
```

See https://vacuumschmelze.com/Company/Quality-and-Safety/Information-for-Manufactured-Articles/VACOFLUX

File: Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/NickelIron/MuMetall.mo [hunk 6]
```diff
@@ -1,12 +1,14 @@
 within Modelica.Magnetic.FluxTubes.Material.SoftMagnetic.NickelIron;
-record MuMetall "MUMETALL (77% NiFe)"
+record MuMetall "Mu-metal (77% NiFe)"
```
Comment (christiankral) [inline Modelica/Magnetic/FluxTubes/Material/SoftMagnetic/NickelIron/MuMetall.mo:2]: In the next major release the record name shall be changed from `MuMetall` to `MuMetal` by a conversion script.

Refs #3995

**Review by christiankral — CHANGES_REQUESTED** [review]
Some of the materials are brand names of the manufacturer Vacuumschmelze. The manufacturer uses the UPPERCASE spelling but in Modelica we have all kinds of different spelling mixed: For the sake of simplicity I propose to keep the used record names in order to avoid changing the record name in the next major release by means of a conversion script. In this case we should at least use the manufacturer name in the label or at least unify the spelling of the labels. 

In the following PNG images the angle `alfa` shall be changed to either `alpha` or the greek letter $\alpha$:

- Modelica/Resources/Images/Magnetic/FluxTubes/Shapes/Circumferential.png
- Modelica/Resources/Images/Magnetic/FluxTubes/Shapes/Toroid.png

**Comment by dietmarw** [comment]
If that bugfix part is really important it should be isolated and not also mixed with all the extensions that this PR now contains. So question is, is it really such a severe bug (which one actually? Ticket?) or can we live with that until the next minor release.

**Comment by AHaumer** [comment]
> If that bugfix part is really important it should be isolated and not also mixed with all the extensions that this PR now contains. So question is, is it really such a severe bug (which one actually? Ticket?) or can we live with that until the next minor release.

I just double-checked in detail:
I've been mistaken by some too complicated formulation of formulas for cross section (area A) and magnetic conductance Gm.
The most annoying topic was that some shapes in QuasiStatic.FluxTubes already had been improved with central angle alpha whereas the corresponding shapes in (transient) FluxTubes had fixed central angle = 2*pi.
Therefore no need for backporting but in my humble opinion improvements.

I'll replace "bug" by "improvement" in the title and the label of this PR.

**Comment by christiankral** [comment]
> In the following PNG images the angle `alfa` shall be changed to either `alpha` or the greek letter :
> 
> * Modelica/Resources/Images/Magnetic/FluxTubes/Shapes/Circumferential.png
> * Modelica/Resources/Images/Magnetic/FluxTubes/Shapes/Toroid.png

This is not yet resolved.

File: Modelica/Magnetic/QuasiStatic/FluxTubes/Shapes/FixedShape/HollowCylinderRadialFlux.mo [hunk 7]
```diff
@@ -13,14 +13,14 @@ model HollowCylinderRadialFlux
...
     annotation (Dialog(group="Fixed geometry"));
-  parameter SI.Angle alpha=2*Modelica.Constants.pi
-  "Central angle";
+  parameter SI.Angle alpha=2*pi "Central angle";
 equation
-  A = l*pi*(r_o + r_i);
+  A = (r_o + r_i)/2*alpha*l;
   // Area at arithmetic mean radius for calculation of average flux density
-  G_m = 2*pi*mu_0*mu_r*l/Modelica.Math.log(r_o/r_i)*alpha/(2*Modelica.Constants.pi);
+  G_m = mu_0*mu_r*alpha*l/Modelica.Math.log(r_o/r_i);
```
Comment (christiankral) [inline Modelica/Magnetic/QuasiStatic/FluxTubes/Shapes/FixedShape/HollowCylinderRadialFlux.mo:20]: This is a bug fix, as the angle `alpha` was not considered correctly until now.

**Review by christiankral — CHANGES_REQUESTED** [review]

File: Modelica/Magnetic/QuasiStatic/FluxTubes/Shapes/FixedShape/HollowCylinderRadialFlux.mo (hunk 7 above)
Comment (dietmarw) [reply Modelica/Magnetic/QuasiStatic/FluxTubes/Shapes/FixedShape/HollowCylinderRadialFlux.mo:20]: So if you think this is important enough to include a hypothetical 4.0.1 release, then this part should probably be moved to its own PR which can then also be backported into the 4.0.x branch.

File: Modelica/Magnetic/FluxTubes/Shapes/FixedShape/HollowCylinderRadialFlux.mo [hunk 8]
```diff
@@ -13,11 +13,12 @@ model HollowCylinderRadialFlux
     annotation (Dialog(group="Fixed geometry"));
   parameter SI.Radius r_o=0.02 "Outer radius of hollow cylinder"
     annotation (Dialog(group="Fixed geometry"));
-
+  parameter SI.Angle alpha=2*pi "Central angle"
+    annotation (Dialog(group="Fixed geometry"));
 equation
-  A = l*pi*(r_o + r_i);
+  A = (r_o + r_i)/2*alpha*l;
```
Comment (christiankral) [inline Modelica/Magnetic/FluxTubes/Shapes/FixedShape/HollowCylinderRadialFlux.mo:19]:
```suggestion
  A = l*alpha*(r_o + r_i)/2;
```

Comment (christiankral) [reply Modelica/Magnetic/FluxTubes/Shapes/FixedShape/HollowCylinderRadialFlux.mo:19]: Improve readability, initiated by https://github.com/modelica/ModelicaStandardLibrary/pull/3997#issuecomment-1160045083

Comment (christiankral) [reply Modelica/Magnetic/FluxTubes/Shapes/FixedShape/HollowCylinderRadialFlux.mo]: Resolved, as committed by 62b02df87d931c6b3464d7794a1a21b1da000003

**Review by christiankral — APPROVED** [review]
OK now

**Pull request merged**